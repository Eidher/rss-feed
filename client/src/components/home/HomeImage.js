import React, { PureComponent } from "react";
import PropTypes from "prop-types";

class HomeImage extends PureComponent {
  static propTypes = {
    image: PropTypes.string.isRequired
  };

  render() {
    return (
      <div className="col-lg-5 my-auto">
        <div className="device-container">
          <div className="device-mockup iphone6_plus portrait white">
            <div className="device">
              <div className="screen">
                <img
                  src={this.props.image}
                  className="img-fluid"
                  alt="Principal"
                />
              </div>
              <div className="button" />
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default HomeImage;
