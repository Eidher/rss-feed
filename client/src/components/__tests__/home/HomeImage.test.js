import React from "react";
import { shallow } from "enzyme";
import HomeImage from "../../home/HomeImage";

describe("<HomeImage />", () => {
  const props = { image: "path-to-image" };
  const wrapper = shallow(<HomeImage {...props} />);

  it("renders correctly.", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("has the right propos ", () => {
    expect(wrapper.find("img")).toHaveProp("src");
    expect(wrapper.find("img").prop("src")).toBe(props.image);
  });
});
