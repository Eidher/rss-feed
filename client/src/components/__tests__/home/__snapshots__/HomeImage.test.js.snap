// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeImage /> renders correctly. 1`] = `
<div
  className="col-lg-5 my-auto"
>
  <div
    className="device-container"
  >
    <div
      className="device-mockup iphone6_plus portrait white"
    >
      <div
        className="device"
      >
        <div
          className="screen"
        >
          <img
            alt="Principal"
            className="img-fluid"
            src="path-to-image"
          />
        </div>
        <div
          className="button"
        />
      </div>
    </div>
  </div>
</div>
`;
